src/v3/recordingComputeUtils.test.ts

```diff
@@ -237,6 +240,431 @@ describe('recordingComputeUtils', () => {
 
       expect(result['operation-span']).toBeDefined()
     })
+
+    describe('matchingIndex', () => {
+      it('should select the correct start span using a positive matchingIndex', () => {
+        const definition: CompleteTraceDefinition<
+          'global',
+          AnyRelation,
+          'origin'
+        > = {
+          ...baseDefinitionFixture,
+          computedSpanDefinitions: {
+            'test-computed-span': {
+              startSpan: Object.assign(
```

Comment:
instead of Object.assign, can we use the `fromDefinition` helper function? same in all the below start/end spans.